### PR TITLE
Pull Request: Implementing Mobile Menu for Portfolio Website

### PR DESCRIPTION
**Pull request by MarcoDDM**
### Description
This pull request implements the mobile menu functionality for the portfolio website. Using JavaScript, the hamburger button will open the mobile menu, and the close button will close it. Additionally, clicking any mobile menu option will navigate the user to the corresponding section of the website. This implementation follows the design guidelines provided in the Figma templates.
### Checklist

- [x] No linter errors in HTML, CSS, and JavaScript
- [x] Correctly followed GitHub flow
- [x] Professional documentation of the work
- [x] Followed best practices for HTML and CSS
- [x] Followed best practices for JavaScript
- [x] Created a separate pull request for the project
- [x] Added the JavaScript + HTML + CSS linter
- [x] Implemented the mobile menu functionality
- [x] Mobile menu opens when clicking the hamburger button
- [x] Mobile menu closes when clicking the close button
- [x] Mobile menu options navigate to correct sections of the website
- [x] Design of the mobile menu follows the Figma templates as closely as possible

Thank you for review my code. Dont hesitate to ask for changes if needed.

### CODE REVIEW

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hi @MarcoDDM 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request 🔴 

You have submitted a project, that does  not have passing linters

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: portfolio.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+});
+
+aboutMe.addEventListener('click', () => {
+  mobileMenu.style.display = 'none';
+  const elements = document.querySelectorAll('.logo, .hamburger, .headline, .works, .about, .form');
+
+  elements.forEach((element) => {
+    element.style.display = 'flex';
+  });
+});
```
Comment (anagudelogu) [inline portfolio.js]: - [ ] I see that you are hiding all the content on your page when the menu is open, however, a better approach would be to set the mobile menu container position property as `absolute`, getting it out of the normal document flow, then styling it with 100vh height and 100vw width, covering the entire screen, and then prevent the scrolling in the body, something like this:

```css
.noScroll {
  overflow: hidden
}

.mobile-menu {
  display: none
  position: absolute;
  height: 100vh;
  width: 100vw;
  top: 0;
  left: 0;
}

.mobile-menu.active {
  display: flex
}
``` 

```js
hamburger.addEventListener('click', ()=>{
  mobileMenu.classList.add('active') // This activates the display: flex in CSS.
  document.body.classList.add('noScroll') // This disables the scrolling on the body when the menu is open.
}

close.addEventListener('click', ()=>{
  mobileMenu.classList.remove('active') // This hides the menu
  document.body.classList.remove('noScroll') // This enables the scrolling on the body again.
}

// Also, you can give a common class to each of the mobile menu links (such as menu-link)
// and you can select all of them avoiding repetition

const menuLinks = querySelectorAll('.menu-link')

// Iterate over the resulting array of links:

menuLinks.forEach((link) => {
  link.addEventListener('click', () => {
      mobileMenu.classList.remove('active') // This hides the menu
      document.body.classList.remove('noScroll') // This enables the scrolling on the body again.
  }
}

```

So you don't actually need to hide and show all elements on your page, you can just create this effect by styling your menu overlaying the other elements, and preventing the scrolling, so only the user interaction can close it! Please consider refactoring your code and removing repetition, remember the DRY rule! 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @MarcoDDM @luisrpavanelli , 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of GitHub flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Great UI design. 💻 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: portfolio.js (hunk 1 above)
Comment (MarcoDDM) [reply portfolio.js]: Oh, thank you! That is a more elegant way to do it. I will implement those changes tomorrow.

**Review by GedeonTS — APPROVED** [review]
# STATUS: APPROVED :tada: :clap: :champagne: 
Hi @MarcoDDM ,

You are amazing :smiley: :clap: !  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

 

## Highlights :clap: :tophat: 
- No linter errors :+1: 
- Good use of GitHub flow :+1: 
- Descriptive PR message :+1: 
- Good use of GitHub :+1: 
## Optional suggestions :small_airplane: 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**